table/server/underdb/glue/src/main/java/alluxio/table/under/glue/GlueDatabase.java

```diff
@@ -0,0 +1,389 @@
+            .addAllDataCols(GlueUtils.toProto(table.getStorageDescriptor().getColumns()))
+            .setStorage(GlueUtils.toProto(table.getStorageDescriptor(), pathTranslator))
+            .setPartitionName(tableName)
+            .putAllParameters(table.getParameters());
+        udbPartitions.add(new GluePartition(
+            new HiveLayout(partitionInfoBuilder.build(), Collections.emptyList())));
+      } else {
+        for (Partition partition : partitions) {
+          String partName = GlueUtils.makePartitionName(partitionColumns, partition.getValues());
+          PartitionInfo.Builder pib = PartitionInfo.newBuilder()
+              .setDbName(getUdbContext().getDbName())
+              .setTableName(tableName)
+              .addAllDataCols(GlueUtils.toProto(partition.getStorageDescriptor().getColumns()))
+              .setStorage(GlueUtils.toProto(partition.getStorageDescriptor(), pathTranslator))
+              .setPartitionName(partName)
+              .putAllParameters(partition.getParameters());
+          if (partition.getValues() != null) {
+            pib.addAllValues(partition.getValues());
+          }
+          udbPartitions.add(new GluePartition(new HiveLayout(pib.build(),
+              statsMap.getOrDefault(partName, Collections.emptyList()))));
+        }
+      }
+
+      return new GlueTable(this,
+          pathTranslator,
+          tableName,
+          GlueUtils.toProtoSchema(table.getStorageDescriptor().getColumns()),
+          columnStatisticsData,
+          // Glue does not provide FieldSchema from API directly
+          // Get FieldSchema from partition keys
+          GlueUtils.toProto(table.getPartitionKeys()),
+          udbPartitions,
+          layout,
+          table);
+    } catch (EntityNotFoundException e) {
+      throw new NotFoundException("Table " + tableName + " does not exist.", e);
+    } catch (ValidationException e) {
+      e.printStackTrace();
+      throw new IOException("Failed to get table: "
+          + tableName + " with validation error: " + e.getMessage(), e);
+    } catch (GlueEncryptionException e) {
+      throw new IOException("Failed to get table: " + tableName + " error: " + e.getMessage(), e);
+    }
+  }
+
+  private List<Partition> batchGetPartitions(AWSGlueAsync glueClient, String tableName)
+      throws IOException {
+    List<Partition> partitions = new ArrayList<>();
+    try {
+      GetPartitionsRequest getPartitionsRequest =
+          new GetPartitionsRequest()
+              .withCatalogId(mGlueConfiguration.get(Property.CATALOG_ID))
+              .withDatabaseName(mGlueDbName)
+              .withTableName(tableName);
+      if (glueClient.getPartitions(getPartitionsRequest).getPartitions() != null) {
+        partitions = glueClient.getPartitions(getPartitionsRequest).getPartitions();
+      }
+      return partitions;
+    } catch (AWSGlueException e) {
+      throw new IOException("WARNING: Cannot get partition information for table: "
```

Comment:
You can remove "WARNING", and also add the catalog id and the database name.

Reply:
Fixed.